vmm/src/device_tree.rs

```diff
@@ -16,7 +17,7 @@ pub struct DeviceNode {
     pub children: Vec<String>,
     #[serde(skip)]
     pub migratable: Option<Arc<Mutex<dyn Migratable>>>,
-    pub pci_bdf: Option<u32>,
+    pub pci_bdf: Option<String>,
```

Comment:
What about using the `PciBdf` type and ensure that it implements `Display`?

Reply:
The problem isn't about display, it's about `Serialize`/`Deserialize`.
I thought it would be simpler to store a `String` rather than manually implementing `Serialize`/`Deserialize` for `PciBdf`.

Reply:
No it's better to store it in the machine friendly format as it makes the comparisons easier and convert only when it needs to be shown.

Reply:
Ok so going down the manual Serialize/Deserialize path?

Reply:
You can just use `Display` and `FromStr` in the implementations.

Reply:
Alright!